src/core/mac/mac.cpp

```diff
@@ -1369,6 +1369,14 @@ void Mac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, otError aError
 
             mBroadcastTransmitCount = 0;
         }
+
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+        // Verify Enh-ACK integrity by checking its MIC
+        if ((aError == OT_ERROR_NONE) && (aAckFrame != nullptr))
+        {
+            aError = (ProcessEnhAckSecurity(aFrame, *aAckFrame) == OT_ERROR_NONE) ? OT_ERROR_NONE : OT_ERROR_NO_ACK;
+        }
```

Comment:
The code as is looks fine. A suggestion below to combine the check in `if`?
```suggestion
        if ((aError == OT_ERROR_NONE) && (aAckFrame != nullptr) && (ProcessEnhAckSecurity(aFrame, *aAckFrame) != OT_ERROR_NONE))
        {
            aError = OT_ERROR_NO_ACK;
        }
```